integration-tests/tests/src/tests/sync/user.ts

```diff
@@ -0,0 +1,620 @@
+    });
+  });
+  describe("custom functions", () => {
+    importAppBefore("with-custom-function");
+    it("custom confirmation function works", async function (this: Mocha.Context & AppContext & RealmContext) {
+      const pendingEmail = randomPendingVerificationEmail();
+      const validPassword = "password123456";
+
+      // we should be able to register our user as pending confirmation
+      await this.app.emailPasswordAuth.registerUser({ email: pendingEmail, password: validPassword });
+
+      // we should be able to call the registration function again
+      await this.app.emailPasswordAuth.retryCustomConfirmation({ email: pendingEmail });
+    });
+    it("reset password function works", async function (this: Mocha.Context & AppContext & RealmContext) {
+      const validEmail = randomVerifiableEmail();
+      const validPassword = "password123456";
+      await this.app.emailPasswordAuth.registerUser({ email: validEmail, password: validPassword });
+
+      const newPassword = "realm_tests_do_reset654321";
+      await this.app.emailPasswordAuth.callResetPasswordFunction({ email: validEmail, password: newPassword });
+
+      // see if we can log in
+      const creds = Realm.Credentials.emailPassword(validEmail, newPassword);
+      const user = await this.app.logIn(creds);
+      expect(user instanceof Realm.User).to.be.true;
+
+      await user.logOut();
+    });
+    it("arbitrary custom function works", async function (this: Mocha.Context & AppContext & RealmContext) {
+      const credentials = Realm.Credentials.anonymous();
+      const user = await this.app.logIn(credentials);
+
+      expect(await user.callFunction("sumFunc")).equals(0);
+      expect(await user.callFunction("sumFunc", 123)).equals(123);
+      expect(await user.functions.sumFunc(123)).equals(123);
+      expect(await user.functions["sumFunc"](123)).equals(123);
+
+      // Test method stashing / that `this` is bound correctly.
+      const sumFunc = user.functions.sumFunc;
+      expect(await sumFunc()).equals(0);
+      expect(await sumFunc(123)).equals(123);
+      expect(await sumFunc(123)).equals(123); // Not just one-shot
+
+      expect(await user.functions.sumFunc()).equals(0);
+      expect(await user.functions.sumFunc(1, 2, 3)).equals(6);
+
+      try {
+        await user.functions.error();
+      } catch (err: any) {
+        expect(err.message).equals("function not found: 'error'");
+      }
+    });
+  });
+  describe("mongo client", () => {
+    importAppBefore("with-db");
+    it("can perform operations on a collection via the client", async function (this: Mocha.Context &
+      AppContext &
+      RealmContext) {
+      const credentials = Realm.Credentials.anonymous();
+      const user = await this.app.logIn(credentials);
+
+      const mongo = user.mongoClient("mongodb");
+      //@ts-expect-error TYPEBUG: serviceName is a missing property on MongoDB interface
+      expect(mongo.serviceName).equals("mongodb");
```

Comment:
This should be fixed with the new SDK: https://github.com/realm/realm-js/blob/bindgen/packages/realm/src/app-services/MongoClient.ts#L166-L169
🤞